Test/ApiFunctional/Model/Checkout/GetQuoteTest.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Amwal\Payments\Test\ApiFunctional\Model\Checkout;
+
+use Magento\TestFramework\TestCase\WebapiAbstract;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\Framework\Webapi\Rest\Request;
+
+class GetQuoteTest extends WebapiAbstract
+{
+    private const SERVICE_VERSION = 'V1';
+    private const SERVICE_NAME = 'Amwal';
+    private const RESOURCE_PATH = '/V1/amwal/get-quote';
+
+    /**
+     * @var \Magento\TestFramework\ObjectManager
+     */
+    protected $objectManager;
+
+
+    protected function setUp(): void
+    {
+        $this->objectManager = Bootstrap::getObjectManager();
+    }
+
+    /**
+     * Test getQuote
+     */
+    public function testGetQuote()
+    {
+        $tempData = require __DIR__ . '../../../_files/TempData.php';
+
+        $serviceInfoForAmwalCart = [
+            'rest' => [
+                'resourcePath' => '/V1/amwal/button/cart',
+                'httpMethod' => Request::HTTP_METHOD_POST,
+            ],
+            'soap' => [
+                'service' => self::SERVICE_NAME,
+                'serviceVersion' => self::SERVICE_VERSION,
+                'operation' => 'AmwalGetCartButtonConfig',
+            ],
+        ];
+
+        $requestData = [
+            'refIdData' => [
+                'identifier' => '100',
+                'customer_id' => '0',
+                'timestamp' => '1707916143'
+            ],
+            'triggerContext' => 'product-detail-page',
+            'locale' => 'en',
+            'cartId' => $tempData['cartId'],
+        ];
+
+        $cartData = $this->_webApiCall($serviceInfoForAmwalCart, $requestData);
+        $this->assertNotEmpty($cartData);
+
+        $requestData = [
+            'merchantId' => $cartData['merchant_id'],
+            'amount' => $cartData['amount'],
+            'taxes' => 0,
+            'discount' => $cartData['discount'],
+            'fees' => 0,
+            'client_email' => 'mardi@amwal.tech',
+            'client_first_name' => 'Ahmed',
+            'client_last_name' => 'Mardi',
+            'client_phone_number' => '+201033233462',
+            'order_details' => [
+                'order_position' => 'product-detail-page',
+                'plugin_version' => 'Magento 1.0.32'
+            ],
+            'address_details' => [
+                'city' => 'Cairo',
+                'state' => 'Cairo',
+                'postcode' => '4472001',
+                'country' => 'EG',
+                'street1' => 'El-Thawra Street Sheraton Al Matar',
+                'street2' => '',
+                'email' => 'mardi@amwal.tech'
+            ],
+            'refId' => $tempData['refId'],
+            'uniqueRef' => false
+        ];
+
+        $transactionData = $this->_executeCurl('https://qa-backend.sa.amwal.tech/transactions/', $requestData);
+        $this->assertNotEmpty($transactionData);
```

Comment:
instead of creating a new transaction from the backend, can you reuse `$transactionData` from an existing one?